depends: add *FLAGS to gen_id

fanquake opened the pull request:
The depends cache should be busted when flags change, the same as any other tooling change. I'd also like to start passing `*FLAGS` into depends inside the Guix env, which, without this change, doesn't bust the cache.

fanquake committed:
depends: add *FLAGS to gen_id

The depends cache should be busted when flags change, the same as any
other tooling change. Id also like to start passing *FLAGS into depends
inside the Guix env, which, without this change, doesn't bust the cache.

hebasto commented:
Concept ACK.

fanquake commented:
Two other things that should happen at the same time as this are adding the linker (this is somewhat compensated for via adding the c/xx flags), as well as making the flag overriding work correctly.

hebasto commented:
My Guix build:
```
aarch64
6361934f5c9bd884aff2ef18000ca718974d8ea564939db6acf3f3ac6faa5e35  guix-build-cd048e03e258/output/aarch64-linux-gnu/SHA256SUMS.part
d3be037f7cd976b315f96303a56952e3eb5bb8f1963d01c0ede701af2cfee83a  guix-build-cd048e03e258/output/aarch64-linux-gnu/bitcoin-cd048e03e258-aarch64-linux-gnu-debug.tar.gz
7c6e75134c8b6614758c024b7bdfcd3bddd0b652d374cbb852e365f00b360a18  guix-build-cd048e03e258/output/aarch64-linux-gnu/bitcoin-cd048e03e258-aarch64-linux-gnu.tar.gz
4514a22fbffe658ef632b013941143c0778b5bacb54a363f4310d3277577b5f0  guix-build-cd048e03e258/output/arm-linux-gnueabihf/SHA256SUMS.part
64ec808aba524f7630cb3b2c3a6944e9b88d72daefed60fbae2f9ddedae7f077  guix-build-cd048e03e258/output/arm-linux-gnueabihf/bitcoin-cd048e03e258-arm-linux-gnueabihf-debug.tar.gz
fa9ba95096142799b4febfee788959fd5a425be0c846dc345c65b0dbfe1a219f  guix-build-cd048e03e258/output/arm-linux-gnueabihf/bitcoin-cd048e03e258-arm-linux-gnueabihf.tar.gz
2356c77fe389353d8e13f961ffb49ad7130a084ff782589e1159677e0a330c5e  guix-build-cd048e03e258/output/arm64-apple-darwin/SHA256SUMS.part
bc501aeb65d96defe117181a53d912fd381b60c7df85fbcf54b375a0f7200f5b  guix-build-cd048e03e258/output/arm64-apple-darwin/bitcoin-cd048e03e258-arm64-apple-darwin-unsigned.tar.gz
3ac2543266223d53cd050bb5fd1394906e64b495ebeef1419763362e3645f55c  guix-build-cd048e03e258/output/arm64-apple-darwin/bitcoin-cd048e03e258-arm64-apple-darwin-unsigned.zip
6fa6115d33fe9387c534d9f898b4bc4320125a74dfac349545ed0ea1d5215d05  guix-build-cd048e03e258/output/arm64-apple-darwin/bitcoin-cd048e03e258-arm64-apple-darwin.tar.gz
ff6c4ccefcfa783457a5cde35289e77a732bb20ef1052f7e38b18608d2974d94  guix-build-cd048e03e258/output/dist-archive/bitcoin-cd048e03e258.tar.gz
ffd97825fdd6c81eb0f10d283b9a5d50ec1076f8cabc70634e3421ca783a4466  guix-build-cd048e03e258/output/powerpc64-linux-gnu/SHA256SUMS.part
bf91ae047f7941af0de90dcd19720fee5ddb52badcadbb54de08fdb2b66a952f  guix-build-cd048e03e258/output/powerpc64-linux-gnu/bitcoin-cd048e03e258-powerpc64-linux-gnu-debug.tar.gz
b320cd3d24fbea6e18a872a3e96814039f32311bff8206fe23231f45d22fceb9  guix-build-cd048e03e258/output/powerpc64-linux-gnu/bitcoin-cd048e03e258-powerpc64-linux-gnu.tar.gz
fb160f2c2333cac19c89a48ee8c9ec67638cda2dae1449520f2fd5186b8cf007  guix-build-cd048e03e258/output/riscv64-linux-gnu/SHA256SUMS.part
6e4103f2b8467f049d9f7bb849fdfd96e04af2b8c4d48c11e2b1fa1e77c2db67  guix-build-cd048e03e258/output/riscv64-linux-gnu/bitcoin-cd048e03e258-riscv64-linux-gnu-debug.tar.gz
18ec18d121cac286ec2f1f0159841bbdac04b1c3bcc2f820b381db73d063b63b  guix-build-cd048e03e258/output/riscv64-linux-gnu/bitcoin-cd048e03e258-riscv64-linux-gnu.tar.gz
15b48d85bbe68a29ed5ccc27212b0afc03e7aa4e1ae84490dba3deeae693aab2  guix-build-cd048e03e258/output/x86_64-apple-darwin/SHA256SUMS.part
c3bdc254e4d3e25514e73c8959e2effb28c16422a447fcd324739de77c384dad  guix-build-cd048e03e258/output/x86_64-apple-darwin/bitcoin-cd048e03e258-x86_64-apple-darwin-unsigned.tar.gz
e25add8b347ebf77acff1548ac14a9acab870efefa557c598366a275c0653687  guix-build-cd048e03e258/output/x86_64-apple-darwin/bitcoin-cd048e03e258-x86_64-apple-darwin-unsigned.zip
e7d43cfc0bcad5e4d6b8b9d3c40262bef95a4c749c5f6db3211c422f421e9ccd  guix-build-cd048e03e258/output/x86_64-apple-darwin/bitcoin-cd048e03e258-x86_64-apple-darwin.tar.gz
7690e77c3948e216bb06bf3205f9be28cac4594b6f187ba98da40b398e6792b3  guix-build-cd048e03e258/output/x86_64-linux-gnu/SHA256SUMS.part
cf1fc1da25f4f82de785294df3957ed967893fa1a77b715763fc5adcd3bc7c69  guix-build-cd048e03e258/output/x86_64-linux-gnu/bitcoin-cd048e03e258-x86_64-linux-gnu-debug.tar.gz
6958cb04755c26f0c3fd44599c00b4b5d14e97897e1177bb4994f5371e76c395  guix-build-cd048e03e258/output/x86_64-linux-gnu/bitcoin-cd048e03e258-x86_64-linux-gnu.tar.gz
f9b87fcaf34cf0182ad620d57b178c4decd562d165944da87593391e7589c24f  guix-build-cd048e03e258/output/x86_64-w64-mingw32/SHA256SUMS.part
a74fadfd30da6af3764191ce910bc1eb8ac9c3a8d8d46a2c5e9e194cc811b1e0  guix-build-cd048e03e258/output/x86_64-w64-mingw32/bitcoin-cd048e03e258-win64-debug.zip
16347d423ed3d83a487fd3d81cef30b1c8998b9c0986ebb1b574f30f7d1a39a0  guix-build-cd048e03e258/output/x86_64-w64-mingw32/bitcoin-cd048e03e258-win64-setup-unsigned.exe
b5e493fbbfc3da13a8969402ccfc7bc8146042bbaa93aee6f12dbbad00698fb6  guix-build-cd048e03e258/output/x86_64-w64-mingw32/bitcoin-cd048e03e258-win64-unsigned.tar.gz
e92c1f39f88d414bba6a04ce0b3fd53f50714ccd463e44f9c1f166091bba8a83  guix-build-cd048e03e258/output/x86_64-w64-mingw32/bitcoin-cd048e03e258-win64.zip
```

hebasto commented on depends/Makefile:
```diff
@@ -142,8 +142,15 @@ include packages/packages.mk
-build_id:=$(shell env CC='$(build_CC)' C_STANDARD='$(C_STANDARD)' CXX='$(build_CXX)' CXX_STANDARD='$(CXX_STANDARD)' AR='$(build_AR) 'NM='$(build_NM)' RANLIB='$(build_RANLIB)' STRIP='$(build_STRIP)' SHA256SUM='$(build_SHA256SUM)' DEBUG='$(DEBUG)' LTO='$(LTO)' NO_HARDEN='$(NO_HARDEN)' ./gen_id '$(BUILD_ID_SALT)' 'GUIX_ENVIRONMENT=$(realpath $(GUIX_ENVIRONMENT))')
-$(host_arch)_$(host_os)_id:=$(shell env CC='$(host_CC)' C_STANDARD='$(C_STANDARD)' CXX='$(host_CXX)' CXX_STANDARD='$(CXX_STANDARD)' AR='$(host_AR)' NM='$(host_NM)' RANLIB='$(host_RANLIB)' STRIP='$(host_STRIP)' SHA256SUM='$(build_SHA256SUM)' DEBUG='$(DEBUG)' LTO='$(LTO)' NO_HARDEN='$(NO_HARDEN)' ./gen_id '$(HOST_ID_SALT)' 'GUIX_ENVIRONMENT=$(realpath $(GUIX_ENVIRONMENT))')
+build_id:=$(shell env CC='$(build_CC)' C_STANDARD='$(C_STANDARD)' CXX='$(build_CXX)' CXX_STANDARD='$(CXX_STANDARD)' \
+                      CPPFLAGS='$(CPPFLAGS)' CFLAGS='$(CFLAGS)' CXXFLAGS='$(CXXFLAGS)' LDFLAGS='$(LDFLAGS)' \
```
I'm not sure about that as none of the `{CPP,C,CXX,LD}FLAGS` variables is propagated to the native packages:
```
$ cd depends
$ make HOST=arm64-apple-darwin MULTIPROCESS=1 print-native_libmultiprocess_cxxflags CXXFLAGS=-some-fancy-flag
native_libmultiprocess_cxxflags= 
```

theuni replied:
So why bother passing them? I don't see any need to rebuild the native packages when changing host flags?

hebasto commented on depends/gen_id:
```diff
@@ -1,6 +1,7 @@
 #!/usr/bin/env bash
 
 # Usage: env [ CC=... ] [ C_STANDARD=...] [ CXX=... ] [CXX_STANDARD=...] \
+#            [ CPPFLAGS=... ] [CFLAGS=...] [CXXFLAGS=...] [LDFLAGS=...]
```
nit:
```suggestion
#            [ CPPFLAGS=... ] [CFLAGS=...] [CXXFLAGS=...] [LDFLAGS=...] \
```

---

While touching this code, the comment can be improved:
```diff
--- a/depends/gen_id
+++ b/depends/gen_id
@@ -3,7 +3,7 @@
 # Usage: env [ CC=... ] [ C_STANDARD=...] [ CXX=... ] [CXX_STANDARD=...] \
 #            [ CPPFLAGS=... ] [CFLAGS=...] [CXXFLAGS=...] [LDFLAGS=...]
 #            [ AR=... ] [ NM=... ] [ RANLIB=... ] [ STRIP=... ] [ DEBUG=... ] \
-#            [ LTO=... ] [ NO_HARDEN=... ] ./build-id [ID_SALT]...
+#            [ LTO=... ] [ NO_HARDEN=... ] ./gen-id [ID_SALT]...
 #
 # Prints to stdout a SHA256 hash representing the current toolset, used by
 # depends/Makefile as a build id for caching purposes (detecting when the
```

fanquake commented on depends/Makefile:
```diff
@@ -142,8 +142,15 @@ include packages/packages.mk
+build_id:=$(shell env CC='$(build_CC)' C_STANDARD='$(C_STANDARD)' CXX='$(build_CXX)' CXX_STANDARD='$(CXX_STANDARD)' \
+                      CPPFLAGS='$(CPPFLAGS)' CFLAGS='$(CFLAGS)' CXXFLAGS='$(CXXFLAGS)' LDFLAGS='$(LDFLAGS)' \
+                      AR='$(build_AR) 'NM='$(build_NM)' RANLIB='$(build_RANLIB)' STRIP='$(build_STRIP)' SHA256SUM='$(build_SHA256SUM)' \
+                      DEBUG='$(DEBUG)' LTO='$(LTO)' NO_HARDEN='$(NO_HARDEN)' ./gen_id '$(BUILD_ID_SALT)' 'GUIX_ENVIRONMENT=$(realpath $(GUIX_ENVIRONMENT))')
```
Broken.

theuni reviewed (commented):
Concept ACK

fanquake committed:
depends: add mold & ld.lld to gen_id

We use `lld` when cross-compiling for macOS, and it's version should
be tied to LLVM. However someone compiling with GCC and `-fuse-ld=lld`
would not see a cache bust if the LLVM toolchain was updated.

We don't use `mold` directly, but I'm aware of it's usage in
infrastructure, along with depends, used to test the project.

theuni commented on depends/gen_id:
```diff
@@ -58,6 +58,17 @@
+    # be tied to LLVM. However someone compiling with GCC and -fuse-ld=lld
+    # would not see a cache bust if the LLVM toolchain was updated.
+    echo "BEGIN lld"
+    bash -c "ld.lld --version"
```
I wonder if this could be replaced with an actual invocation of the linker instead? Something like:

`bash -c "echo \"int main(){}\" | $CC -v $CFLAGS $LDFLAGS -Wl,-v -xc -o /dev/null -"`

fanquake replied:
We can do something more involved if you like, also want to avoid anything that is likely to be flaky.

theuni replied:
Eh, probably not worth bothering with.

theuni replied:
Mmm, it is a shame that an update to an unused linker would bust the cache. But I guess that'd be so rare that it's not worth being too concerned about.

We can always revisit and switch to an invocation if it turns out to be too greedy.

theuni reviewed (approved):
utACK b31ec088601994d0c59f874bd8e81b43339ef60b

theuni commented on depends/Makefile:
```diff
@@ -141,8 +141,16 @@ include packages/packages.mk
 #
-build_id:=$(shell env CC='$(build_CC)' C_STANDARD='$(C_STANDARD)' CXX='$(build_CXX)' CXX_STANDARD='$(CXX_STANDARD)' AR='$(build_AR) 'NM='$(build_NM)' RANLIB='$(build_RANLIB)' STRIP='$(build_STRIP)' SHA256SUM='$(build_SHA256SUM)' DEBUG='$(DEBUG)' LTO='$(LTO)' NO_HARDEN='$(NO_HARDEN)' ./gen_id '$(BUILD_ID_SALT)' 'GUIX_ENVIRONMENT=$(realpath $(GUIX_ENVIRONMENT))')
-$(host_arch)_$(host_os)_id:=$(shell env CC='$(host_CC)' C_STANDARD='$(C_STANDARD)' CXX='$(host_CXX)' CXX_STANDARD='$(CXX_STANDARD)' AR='$(host_AR)' NM='$(host_NM)' RANLIB='$(host_RANLIB)' STRIP='$(host_STRIP)' SHA256SUM='$(build_SHA256SUM)' DEBUG='$(DEBUG)' LTO='$(LTO)' NO_HARDEN='$(NO_HARDEN)' ./gen_id '$(HOST_ID_SALT)' 'GUIX_ENVIRONMENT=$(realpath $(GUIX_ENVIRONMENT))')
+build_id:=$(shell env CC='$(build_CC)' C_STANDARD='$(C_STANDARD)' CXX='$(build_CXX)' CXX_STANDARD='$(CXX_STANDARD)' \
```
Nit: Seems LTO shouldn't be here either, I'm not sure why we'd need to rebuild the build stuff if target LTO changed. But it was already like that, so no need to change it here.

fanquake replied:
Dropped this out as well.

hebasto commented on depends/Makefile:
```diff
@@ -141,8 +141,16 @@ include packages/packages.mk
-build_id:=$(shell env CC='$(build_CC)' C_STANDARD='$(C_STANDARD)' CXX='$(build_CXX)' CXX_STANDARD='$(CXX_STANDARD)' AR='$(build_AR) 'NM='$(build_NM)' RANLIB='$(build_RANLIB)' STRIP='$(build_STRIP)' SHA256SUM='$(build_SHA256SUM)' DEBUG='$(DEBUG)' LTO='$(LTO)' NO_HARDEN='$(NO_HARDEN)' ./gen_id '$(BUILD_ID_SALT)' 'GUIX_ENVIRONMENT=$(realpath $(GUIX_ENVIRONMENT))')
-$(host_arch)_$(host_os)_id:=$(shell env CC='$(host_CC)' C_STANDARD='$(C_STANDARD)' CXX='$(host_CXX)' CXX_STANDARD='$(CXX_STANDARD)' AR='$(host_AR)' NM='$(host_NM)' RANLIB='$(host_RANLIB)' STRIP='$(host_STRIP)' SHA256SUM='$(build_SHA256SUM)' DEBUG='$(DEBUG)' LTO='$(LTO)' NO_HARDEN='$(NO_HARDEN)' ./gen_id '$(HOST_ID_SALT)' 'GUIX_ENVIRONMENT=$(realpath $(GUIX_ENVIRONMENT))')
+build_id:=$(shell env CC='$(build_CC)' C_STANDARD='$(C_STANDARD)' CXX='$(build_CXX)' CXX_STANDARD='$(CXX_STANDARD)' \
+                      AR='$(build_AR) 'NM='$(build_NM)' RANLIB='$(build_RANLIB)' STRIP='$(build_STRIP)' SHA256SUM='$(build_SHA256SUM)' \
```
It should be
```suggestion
                      AR='$(build_AR)' NM='$(build_NM)' RANLIB='$(build_RANLIB)' STRIP='$(build_STRIP)' SHA256SUM='$(build_SHA256SUM)' \
```

Fixed in https://github.com/bitcoin/bitcoin/pull/31627.

hebasto reviewed (approved):
ACK 01df180bfb82c7eafac4638ced249bee4409784b.

It might be better to place this [comment](https://github.com/bitcoin/bitcoin/pull/31125/files#r1878548250) in a more prominent location, such as the PR description or within a source code comment.

fanquake commented:
Guix Build:
```bash
3922712c31b158aefa34a0700c02dc0c2c02c318002ead81cc7c9836c34c7b6a  guix-build-01df180bfb82/output/aarch64-linux-gnu/SHA256SUMS.part
600a22f791e838113d1b0401f0b038a5cf7e28638e6eb80cd80f39885dd2c63b  guix-build-01df180bfb82/output/aarch64-linux-gnu/bitcoin-01df180bfb82-aarch64-linux-gnu-debug.tar.gz
e9a43319a6f6358940ff265be574d067604ce4f31d7c1e5be0c143c4407892ef  guix-build-01df180bfb82/output/aarch64-linux-gnu/bitcoin-01df180bfb82-aarch64-linux-gnu.tar.gz
a049be13012976871a66923c54a4de0814769f9b7e9b0389ed94e5293e6401bf  guix-build-01df180bfb82/output/arm-linux-gnueabihf/SHA256SUMS.part
81a4dd99cddd275594e0049281e954f8c338325c3d6056bc64cb33c0a300ab32  guix-build-01df180bfb82/output/arm-linux-gnueabihf/bitcoin-01df180bfb82-arm-linux-gnueabihf-debug.tar.gz
1ae3d96227047d9f2120c73669c1557f7cbdc8680fca342a0a75aa559292a7c6  guix-build-01df180bfb82/output/arm-linux-gnueabihf/bitcoin-01df180bfb82-arm-linux-gnueabihf.tar.gz
69ca4cab512ef4db170b2bc98a7f326be8f01fdf8f61b3fb6186e8e89d54d479  guix-build-01df180bfb82/output/arm64-apple-darwin/SHA256SUMS.part
0870040b52fe59bd34c536a9c7feec9211b75dc73482b5523c76108c990f84b4  guix-build-01df180bfb82/output/arm64-apple-darwin/bitcoin-01df180bfb82-arm64-apple-darwin-unsigned.tar.gz
08097ec13c55c485c817b1720833b13fb7c2f68c2fde7ad615fba22234599337  guix-build-01df180bfb82/output/arm64-apple-darwin/bitcoin-01df180bfb82-arm64-apple-darwin-unsigned.zip
8449c7d93ef88e0f29469a06cc84eb4307e74e3619809f25948148b467e20e19  guix-build-01df180bfb82/output/arm64-apple-darwin/bitcoin-01df180bfb82-arm64-apple-darwin.tar.gz
5869047d7291aa691d1b5d5ec62770183ec14336b43422fcf249a3c27dc1d87b  guix-build-01df180bfb82/output/dist-archive/bitcoin-01df180bfb82.tar.gz
0176a202d7f88e1069216d7baa62e75ccd22edd2ae98d392c7e9c911f6ae9c32  guix-build-01df180bfb82/output/powerpc64-linux-gnu/SHA256SUMS.part
fc16a5cf38dacec3044f68ae40ed1afabf3329751b52cafff054fcc5d18dfd71  guix-build-01df180bfb82/output/powerpc64-linux-gnu/bitcoin-01df180bfb82-powerpc64-linux-gnu-debug.tar.gz
fdf6d3e42911f53f41dfef9d7e50207c4f83079523296967370826892ed0df9f  guix-build-01df180bfb82/output/powerpc64-linux-gnu/bitcoin-01df180bfb82-powerpc64-linux-gnu.tar.gz
f919e600052054180fa0f38fddcf470b50db7734732b77f7f56118d2fc9b2252  guix-build-01df180bfb82/output/riscv64-linux-gnu/SHA256SUMS.part
894c33913ae86c7be43863737c1a059731c4d6c27e2d10aba47e8ee7c5101c17  guix-build-01df180bfb82/output/riscv64-linux-gnu/bitcoin-01df180bfb82-riscv64-linux-gnu-debug.tar.gz
79fde5858916e7e24e0b0be2c1ab0c1afbe32b9e2648a0d71579079511962d9c  guix-build-01df180bfb82/output/riscv64-linux-gnu/bitcoin-01df180bfb82-riscv64-linux-gnu.tar.gz
0d0752df2dd955618ee6d083d4a2fa1723c5fcbdf245b27f510e5fdbd65e22ac  guix-build-01df180bfb82/output/x86_64-apple-darwin/SHA256SUMS.part
5eb10ba2f175ce0e544b8d137a4e953cdb78ce6e7184fccd827fb285804c4c13  guix-build-01df180bfb82/output/x86_64-apple-darwin/bitcoin-01df180bfb82-x86_64-apple-darwin-unsigned.tar.gz
bf62fe5359b0b3ceeb33415364914b2862c019c1b348d6ae2fb3af5b88617ae7  guix-build-01df180bfb82/output/x86_64-apple-darwin/bitcoin-01df180bfb82-x86_64-apple-darwin-unsigned.zip
a2b2fb991b625eb9efc2ef8ee7c14971fb7cdef889b679d78dc3e500c5124105  guix-build-01df180bfb82/output/x86_64-apple-darwin/bitcoin-01df180bfb82-x86_64-apple-darwin.tar.gz
1ff9628c86bb2fb3a95ec8315e63baa0a046fae47fd1bd5ca8cb0ad39592cc56  guix-build-01df180bfb82/output/x86_64-linux-gnu/SHA256SUMS.part
6ec645d35a0db281f75bf5e6dd71794ec8dbc699361ad923352dc247705af24e  guix-build-01df180bfb82/output/x86_64-linux-gnu/bitcoin-01df180bfb82-x86_64-linux-gnu-debug.tar.gz
a7b3218dbecfb53ea2a10f5a042602c226cadbe2874be2123a7b87ddf726338f  guix-build-01df180bfb82/output/x86_64-linux-gnu/bitcoin-01df180bfb82-x86_64-linux-gnu.tar.gz
81c51442de9a58971cd256bd8a5c911f3a54c45ea2a9a5b223ed5748f41270d7  guix-build-01df180bfb82/output/x86_64-w64-mingw32/SHA256SUMS.part
5eb50b0b69bf8fadf58834c3d414dd95ccb7100005772c0b91d936fdb487fc33  guix-build-01df180bfb82/output/x86_64-w64-mingw32/bitcoin-01df180bfb82-win64-debug.zip
beffccb03c8ed679320dd1d545103d92c89d030510b6dd86ef5ce8eb86db00e4  guix-build-01df180bfb82/output/x86_64-w64-mingw32/bitcoin-01df180bfb82-win64-setup-unsigned.exe
bfce174d7d50e343f119da29afe9f1ad17d48ef949c03b75cd95bb3ad4250575  guix-build-01df180bfb82/output/x86_64-w64-mingw32/bitcoin-01df180bfb82-win64-unsigned.tar.gz
451344b9ef227e9a0611e809795e60f9b643e155c56435c964c37bc7a0870655  guix-build-01df180bfb82/output/x86_64-w64-mingw32/bitcoin-01df180bfb82-win64.zip
```

hebasto commented:
My Guix build:
```
aarch64
3922712c31b158aefa34a0700c02dc0c2c02c318002ead81cc7c9836c34c7b6a  guix-build-01df180bfb82/output/aarch64-linux-gnu/SHA256SUMS.part
600a22f791e838113d1b0401f0b038a5cf7e28638e6eb80cd80f39885dd2c63b  guix-build-01df180bfb82/output/aarch64-linux-gnu/bitcoin-01df180bfb82-aarch64-linux-gnu-debug.tar.gz
e9a43319a6f6358940ff265be574d067604ce4f31d7c1e5be0c143c4407892ef  guix-build-01df180bfb82/output/aarch64-linux-gnu/bitcoin-01df180bfb82-aarch64-linux-gnu.tar.gz
a049be13012976871a66923c54a4de0814769f9b7e9b0389ed94e5293e6401bf  guix-build-01df180bfb82/output/arm-linux-gnueabihf/SHA256SUMS.part
81a4dd99cddd275594e0049281e954f8c338325c3d6056bc64cb33c0a300ab32  guix-build-01df180bfb82/output/arm-linux-gnueabihf/bitcoin-01df180bfb82-arm-linux-gnueabihf-debug.tar.gz
1ae3d96227047d9f2120c73669c1557f7cbdc8680fca342a0a75aa559292a7c6  guix-build-01df180bfb82/output/arm-linux-gnueabihf/bitcoin-01df180bfb82-arm-linux-gnueabihf.tar.gz
69ca4cab512ef4db170b2bc98a7f326be8f01fdf8f61b3fb6186e8e89d54d479  guix-build-01df180bfb82/output/arm64-apple-darwin/SHA256SUMS.part
0870040b52fe59bd34c536a9c7feec9211b75dc73482b5523c76108c990f84b4  guix-build-01df180bfb82/output/arm64-apple-darwin/bitcoin-01df180bfb82-arm64-apple-darwin-unsigned.tar.gz
08097ec13c55c485c817b1720833b13fb7c2f68c2fde7ad615fba22234599337  guix-build-01df180bfb82/output/arm64-apple-darwin/bitcoin-01df180bfb82-arm64-apple-darwin-unsigned.zip
8449c7d93ef88e0f29469a06cc84eb4307e74e3619809f25948148b467e20e19  guix-build-01df180bfb82/output/arm64-apple-darwin/bitcoin-01df180bfb82-arm64-apple-darwin.tar.gz
5869047d7291aa691d1b5d5ec62770183ec14336b43422fcf249a3c27dc1d87b  guix-build-01df180bfb82/output/dist-archive/bitcoin-01df180bfb82.tar.gz
0176a202d7f88e1069216d7baa62e75ccd22edd2ae98d392c7e9c911f6ae9c32  guix-build-01df180bfb82/output/powerpc64-linux-gnu/SHA256SUMS.part
fc16a5cf38dacec3044f68ae40ed1afabf3329751b52cafff054fcc5d18dfd71  guix-build-01df180bfb82/output/powerpc64-linux-gnu/bitcoin-01df180bfb82-powerpc64-linux-gnu-debug.tar.gz
fdf6d3e42911f53f41dfef9d7e50207c4f83079523296967370826892ed0df9f  guix-build-01df180bfb82/output/powerpc64-linux-gnu/bitcoin-01df180bfb82-powerpc64-linux-gnu.tar.gz
f919e600052054180fa0f38fddcf470b50db7734732b77f7f56118d2fc9b2252  guix-build-01df180bfb82/output/riscv64-linux-gnu/SHA256SUMS.part
894c33913ae86c7be43863737c1a059731c4d6c27e2d10aba47e8ee7c5101c17  guix-build-01df180bfb82/output/riscv64-linux-gnu/bitcoin-01df180bfb82-riscv64-linux-gnu-debug.tar.gz
79fde5858916e7e24e0b0be2c1ab0c1afbe32b9e2648a0d71579079511962d9c  guix-build-01df180bfb82/output/riscv64-linux-gnu/bitcoin-01df180bfb82-riscv64-linux-gnu.tar.gz
0d0752df2dd955618ee6d083d4a2fa1723c5fcbdf245b27f510e5fdbd65e22ac  guix-build-01df180bfb82/output/x86_64-apple-darwin/SHA256SUMS.part
5eb10ba2f175ce0e544b8d137a4e953cdb78ce6e7184fccd827fb285804c4c13  guix-build-01df180bfb82/output/x86_64-apple-darwin/bitcoin-01df180bfb82-x86_64-apple-darwin-unsigned.tar.gz
bf62fe5359b0b3ceeb33415364914b2862c019c1b348d6ae2fb3af5b88617ae7  guix-build-01df180bfb82/output/x86_64-apple-darwin/bitcoin-01df180bfb82-x86_64-apple-darwin-unsigned.zip
a2b2fb991b625eb9efc2ef8ee7c14971fb7cdef889b679d78dc3e500c5124105  guix-build-01df180bfb82/output/x86_64-apple-darwin/bitcoin-01df180bfb82-x86_64-apple-darwin.tar.gz
1ff9628c86bb2fb3a95ec8315e63baa0a046fae47fd1bd5ca8cb0ad39592cc56  guix-build-01df180bfb82/output/x86_64-linux-gnu/SHA256SUMS.part
6ec645d35a0db281f75bf5e6dd71794ec8dbc699361ad923352dc247705af24e  guix-build-01df180bfb82/output/x86_64-linux-gnu/bitcoin-01df180bfb82-x86_64-linux-gnu-debug.tar.gz
a7b3218dbecfb53ea2a10f5a042602c226cadbe2874be2123a7b87ddf726338f  guix-build-01df180bfb82/output/x86_64-linux-gnu/bitcoin-01df180bfb82-x86_64-linux-gnu.tar.gz
81c51442de9a58971cd256bd8a5c911f3a54c45ea2a9a5b223ed5748f41270d7  guix-build-01df180bfb82/output/x86_64-w64-mingw32/SHA256SUMS.part
5eb50b0b69bf8fadf58834c3d414dd95ccb7100005772c0b91d936fdb487fc33  guix-build-01df180bfb82/output/x86_64-w64-mingw32/bitcoin-01df180bfb82-win64-debug.zip
beffccb03c8ed679320dd1d545103d92c89d030510b6dd86ef5ce8eb86db00e4  guix-build-01df180bfb82/output/x86_64-w64-mingw32/bitcoin-01df180bfb82-win64-setup-unsigned.exe
bfce174d7d50e343f119da29afe9f1ad17d48ef949c03b75cd95bb3ad4250575  guix-build-01df180bfb82/output/x86_64-w64-mingw32/bitcoin-01df180bfb82-win64-unsigned.tar.gz
451344b9ef227e9a0611e809795e60f9b643e155c56435c964c37bc7a0870655  guix-build-01df180bfb82/output/x86_64-w64-mingw32/bitcoin-01df180bfb82-win64.zip
```
